### PR TITLE
Testcontainers

### PR DESCRIPTION
**Pull request by ZombieAlienRobot**
## Description of changes
Updated to use testcontainers and cleaned up pipeline.

  
## How has this been tested?
Please describe the test strategy you followed.

- [X] automated unit test
- [X] automated integration test
- [X] automated acceptance test
- [ ] manual, exploratory test
    
## Checklist before requesting a review
- [X] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My code fulfills all acceptance criteria
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [X] I have made corresponding changes to the documentation
- [X] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [X] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -100,4 +100,4 @@ tasks.named('test') {
 	useJUnitPlatform()
```
Comment (PaulBredl) [inline build.gradle:100]: The container library should also be added here, or not?

**Review by PaulBredl — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (ZombieAlienRobot) [reply build.gradle:102]: Yes, added them now

Comment (PaulBredl) [reply build.gradle:102]: oh yeah, I just ment the gradle file in general. 👍

**Review by PaulBredl — APPROVED** [review]
